tidepool_data_science_metrics/cgm/cgm.py

```diff
@@ -1,104 +1,196 @@
         The number of digits to round the result to.
 
     Returns
     -------
     int
         The percent value by range.
     """
-
+    _validate_bg(bg_array)
     calc_low_thresh, calc_upper_thresh = _validate_input(
         lower_threshold, upper_threshold
     )
     results = round(
         np.where(
-            (bg_values <= calc_upper_thresh) & (bg_values >= calc_low_thresh), 1, 0
+            (bg_array <= calc_upper_thresh) & (bg_array >= calc_low_thresh), 1, 0
         ).sum()
-        / bg_values.size
+        / bg_array.size
         * 100,
         round_val,
     )
     return results
 
 
-def percent_time_in_range(
-    bg_values, lower_threshold: int, upper_threshold: int, round_val=2, time_delta=5
-):
+def percent_time_in_range_70_180(
+    bg_array: "np.ndarray[np.int64]", round_val: int = 2
+) -> np.float64:
+    """
+    Calculate the percent of values with a blood glucose value between 70 and 180.
 
-    calc_low_thresh, calc_upper_thresh = _validate_input(
-        lower_threshold, upper_threshold
+    Parameters
+    ----------
+    bg_array : ndarray
+        1D array containing data with `int` type.
+    round_val : int, optional
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The percent value with in the range between 70 and 180.
+    """
+    return percent_values_by_range(
+        bg_array, lower_threshold=70, upper_threshold=180, round_val=round_val
+    )
+
+
+def percent_time_above_180(
+    bg_array: "np.ndarray[np.int64]", round_val: int = 2
+) -> np.float64:
+    """
+    Calculate the percent of values with a blood glucose above 180.
+
+    Parameters
+    ----------
+    bg_array : ndarray
+        1D array containing data with `int` type.
+    round_val : int, optional
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The percent values above 180.
+    """
+    return percent_values_by_range(
+        bg_array, lower_threshold=180, upper_threshold=1000, round_val=round_val
     )
-    bg_df = bg_values["bg_values"]
-    bg_np = bg_df.to_numpy()
-    in_range = np.count_nonzero((bg_np > calc_low_thresh) & (bg_np < calc_upper_thresh))
-    val_count = bg_df.count()
-    return round(in_range / val_count * 100, round_val)
 
 
-def episodes(bg_values_df, episodes_threshold: int, min_ct_per_ep=3, min_duration=5):
+def percent_time_below_70(
+    bg_array: "np.ndarray[np.int64]", round_val: int = 2
+) -> np.float64:
+    """
+    Calculate the percent of values with a blood glucose below 70.
+
+    Parameters
+    ----------
+    bg_array : ndarray
+        1D array containing data with `int` type.
+    round_val : int, optional
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The percent values below 70.
+    """
+    return percent_values_by_range(
+        bg_array, lower_threshold=0, upper_threshold=70, round_val=round_val
```

Comment:
69.499 instead of 70, and I think the lower threshold needs to be 1 instead of 0.